python/ray/_raylet.pxd

```diff
@@ -106,6 +107,7 @@ cdef class CoreWorker:
                             CObjectID *c_object_id, shared_ptr[CBuffer] *data,
                             c_bool created_by_worker,
                             owner_address=*)
+    cdef shared_ptr[CAddress] _convert_python_address(self, address=*)
```

Comment:
why do we need shared ptr here?

Reply:
I use shared ptr to replace `absl::optional` or `boost::optional`

Reply:
What I mean is that will unique_ptr also work here? Ideally, we shouldn't use shared ptr unless it's shared across threads.

Reply:
I got it, I'll change it to unique_ptr